develop/networking.md

```diff
@@ -26,69 +26,28 @@ so this page is split up into a few categories.
-        BlockPos target = ...
-
-        // BAD CODE: DON'T EVER DO THIS! // [!code error]
-        ClientBlockHighlighting.highlightBlock(MinecraftClient.getInstance(), target); // [!code error]
-        return super.use(world, user, hand);
-    }
-}
-```
-
-Upon testing, you will see a lightning bolt being summoned and nothing crashes. Now you want to show the mod to your
-friend, you boot up a dedicated server and invite your friend on with the mod installed.
-
-You use the item and the server crashes. You will probably notice in the crash log an error similar to this:
-
-```log
-[Server thread/FATAL]: Error executing task on Server
-java.lang.RuntimeException: Cannot load class net.minecraft.client.MinecraftClient in environment type SERVER
-```
-
-### Why Does the Server Crash? {#why-does-the-server-crash}
-
-The code calls logic only present on the client distribution of the Minecraft. The reason for Mojang distributing the
-game in this way is to cut down on the size of the Minecraft Server JAR file. There isn't really a reason to include an
-entire rendering engine when your own machine will render the world.
+One imporant thing to keep in mind is even in single player there is still a server running, Packets are still used to communicate between
+the client and server even when no one else is playing with you. We usually call this the logicial client/server or the client/server thread.
+This is diffrent from the client/server code split, that prevents you from being able to use client only classes in common (shared between
+the client and server) code.
 
-In a development environment, client-only classes are indicated by the `@Environment(EnvType.CLIENT)` annotation.
-
-### How Do We Fix the Crash? {#how-do-we-fix-the-crash}
-
-To fix this issue, you need to understand how the game client and dedicated server communicate:
-
-![Sides](/assets/develop/networking/sides.png)
-
-The diagram above shows that the game client and dedicated server are separate systems, bridged together using
-_packets_. Packets can contain data which we refer to as the _payload_.
-
-This packet bridge does not only exist between a game client and dedicated server, but also between your client and
-another client connected over LAN. The packet bridge is also present even in singleplayer. This is because the game
-client will spin up a special integrated server instance to run the game on.
-
-Connection to a server over LAN or singleplayer can also be treated like the server is a remote, dedicated server; so
-your game client can't directly access the server instance.
+When state is not synced between the client and server, you can run into issues where the server or other clients dont agree with what the
+client is doing. This is often known as a "desync". When writing your own mod you may need to send a packet of data to notify the server,
+client or any other connected clients that something has happened.
```

Comment:
It is the server's job to notify other clients about changes. Merging "notify server (from client)" and "notify client (from server)" in the same sentence like this is confusing